### PR TITLE
Contact form

### PR DESCRIPTION
**Pull request by CorneliaKelinske**
@alanvardy 

I have added testing for my contact controller.
I shifted the test for whether my contact link leads to the contact page to my page controller.
And I changed the mailer test so as to reflect the valid params that I also use in the contact controller test.
 
It would also be great if you could read my first blog post while you are at it. 
All in all, I think I am in a place where I am comfortable with the idea of deploying this site.

Thank you!

### CODE REVIEW

File: lib/first_blog_web/controllers/contact_controller.ex [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      # Failed changeset validation
+      {:error, %Ecto.Changeset{} = changeset} ->
+        conn
+        |> put_flash(:error, "There was a problem sending your message")
+        |> render("new.html", changeset: changeset)
+
+      # Other error
+      _error ->
+        conn
+        |> put_flash(:error, "Ouuups")
```
Comment (alanvardy) [inline lib/first_blog_web/controllers/contact_controller.ex]: Maybe include "the email failed to send" as the user might be wondering if they typed something incorrectly.

File: lib/first_blog_web/controllers/contact_controller.ex [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        |> redirect(to: Routes.contact_path(conn, :new))
+    end
+  end
+
+  defp new_changeset, do: Contact.changeset(%{})
```
Comment (alanvardy) [inline lib/first_blog_web/controllers/contact_controller.ex]: I would remove this and just put `Contact.changeset(%{})` on line 9

**Review by alanvardy — APPROVED** [review]
Looking good!

**Comment by alanvardy** [comment]
I can show you how to squash down all those commits if you like too.

**Comment by CorneliaKelinske** [comment]
@alanvardy How about the blog post? Did you like it?

**Pull request merged**